CRM/Profile/Page/Dynamic.php

```diff
@@ -314,7 +326,10 @@ public function run() {
 
         $labels[$index] = preg_replace('/\s+|\W+/', '_', $name);
       }
-
+      $this->isShowEmailTaskLink = 1;
+      if ($this->isShowEmailTaskLink && !empty($fields['email-Primary'])) {
```

Comment:
@eileenmcnaughton the code is not working as expected because it's harcoded to the primary email. And even for the primary email is not working. For some reason the primary email field name for me it's `email-1` instead of `email-Primary`. This could be because we changed the fields in the contact's card profile.

Regardless, this solution doesn't work for all emails, just the primary one. It would be good if it supported all emails.